core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -1051,6 +1049,15 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
     val jobAttemptId = newTaskAttemptID(jobtrackerID, stageId, isMap = true, 0, 0)
     val jobTaskContext = newTaskAttemptContext(wrappedConf.value, jobAttemptId)
     val jobCommitter = jobFormat.getOutputCommitter(jobTaskContext)
+
+    // If speculation is enabled, we only allow FileOutputCommitter.
+    val speculationEnabled = self.conf.getBoolean("spark.speculation", false)
+    if (speculationEnabled && jobCommitter.getClass !=
+      classOf[org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter]) {
+      throw new SparkException(s"Cannot use ${jobCommitter.getClass} as output committer when" +
+        "speculation is enabled.")
+    }
+
```

Comment:
`jobFormat.getOutputCommitter()` will return the output committer associated with the output format. Basically, in normal cases, you cannot specify output committer for `mapreduce` api (the new api). So, I think we should not make change at here. Also, `jobCommitter.getClass != classOf[org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter]` is a strong condition, which is not always the case. Every output format implemented with `mapreduce` API can have its own output committer (e.g. `org.apache.parquet.hadoop.ParquetOutputCommitter`).

Let's leave this part unchanged.